scripts/prepare-subcharts

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env bash
+set -e
+
+# Pull in the subcharts required by a chart and apply any patches on subcharts with fixed version numbers
+
+if [[ -z $1 ]]; then
+	echo "No directory provided to pull in subcharts"
+	exit 1
+fi
+
+f=$1
+
+if ! [[ -f ${f}/charts/requirements.yaml ]]; then
+	echo "Chart ${CHART} has no dependencies; skipping step to pull in dependencies"
+	exit 0
+fi
+
+# Apply the patch on the requirements.yaml
```

Comment:
Charts like `rancher-monitoring` that introduce new subcharts like Prometheus Adapter [as part of the patch](https://github.com/rancher/charts/blob/6293280499af87bbe4905c15d8e938d1c273ac4e/packages/rancher-monitoring/rancher-monitoring.patch#L291-L384) need to be able to pull in the added dependencies as part of preparing the subcharts; therefore, `requirements.yaml` needs to be patched before running the prepare on the subcharts.

An initial solution I had was to just run the entire patch first before trying to pull in dependencies, but there's an edge case in the following scenario:
1. I run `make prepare` on the chart with the current set of scripts
1. I make some patches to the added subcharts (i.e. Prometheus Adapter) and run `make patch` to generate the patch. Patch is successfully generated
1. I run `make prepare` again and the patch is rejected since, before any dependencies are pulled in, the upstream chart has no folder for my added subchart (i.e. `charts/charts/prometheus-adapter` does not exist). Therefore, it can't apply a patch to my non-existent subchart and fails.

It seemed like the easiest way to resolve this was to just apply any patches required on the `requirements.yaml` only, pull in the subcharts (so `charts/charts/prometheus-adapter` now exists), unapply the `reqiuirements.yaml`-only patch, and then continue the `prepare` process normally to apply the patch as expected.